x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/form.tsx

```diff
@@ -378,17 +378,36 @@ export function useOnSubmit({
 
       const hasGoogleCloudShell = data?.item ? getCloudShellUrlFromPackagePolicy(data.item) : false;
 
-      if (agentCount > 0) {
+      // Check if agentless is configured in ESS and Serverless until Agentless API migrates to Serverless
+      const isAgentlessConfigured =
+        isAgentlessAgentPolicy(createdPolicy) || isAgentlessPackagePolicy(data!.item);
+
+      // Removing this code will disabled the Save and Continue button. We need code below update form state and trigger correct modal depending on agent count
+      if (hasFleetAddAgentsPrivileges && !isAgentlessConfigured) {
+        if (hasAzureArmTemplate) {
```

Comment:
it would be easier to read with something like this and avoid some useless setFormState

```typescript
       if (hasAzureArmTemplate) {
          setFormState(agentCount ? 'SUBMITTED' : 'SUBMITTED_AZURE_ARM_TEMPLATE');
        } else if (hasCloudFormation) {
          setFormState(agentCount ? 'SUBMITTED' : 'SUBMITTED_CLOUD_FORMATION');
        } else if(...) {
        } else {
          setFormState(agentCount ? 'SUBMITTED' : 'SUBMITTED_NO_AGENTS');
        }
```